packages/@tailwindcss-upgrade/src/utils/extract-static-plugins.test.ts

```diff
@@ -67,68 +84,204 @@ describe('findStaticPlugins', () => {
+        import plugin2 from './plugin2'
 
-      export default {
-        plugins: [plugin1, () => {} ]
-      }
+        export default {
+          plugins: [
+            plugin1({
+              foo: 'bar',
+            }),
+            plugin2(),
+            require('./plugin3')({
+              foo: 'bar',
+            }),
+          ]
+        }
     `),
-    ).toEqual(null)
+    ).toEqual([
+      ['./plugin1', { foo: 'bar' }],
+      ['./plugin2', null],
+      ['./plugin3', { foo: 'bar' }],
+    ])
+  })
 
+  test('can extract all supported data types', () => {
     expect(
       findStaticPlugins(js`
-      let plugin1 = () => {}
+        import plugin from 'plugin'
 
-      export default {
-        plugins: [plugin1]
-      }
+        export default {
+          plugins: [
+            plugin({
+              'is-arr-mixed': [null, true, false, 1234567, 1.35, 'foo', 'bar', 'true'],
+              'is-arr': ['foo', 'bar'],
+              'is-null': null,
+              'is-true': true,
+              'is-false': false,
+              'is-int': 1234567,
+              'is-float': 1.35,
+              'is-sci': 1.35e-5,
+              'is-str-null': 'null',
+              'is-str-true': 'true',
+              'is-str-false': 'false',
+              'is-str-int': '1234567',
+              'is-str-float': '1.35',
+              'is-str-sci': '1.35e-5',
+            }),
+          ]
+        }
     `),
+    ).toEqual([
+      [
+        'plugin',
+        {
+          'is-arr-mixed': [null, true, false, 1234567, 1.35, 'foo', 'bar', 'true'],
+          'is-arr': ['foo', 'bar'],
+          'is-null': null,
+          'is-true': true,
+          'is-false': false,
+          'is-int': 1234567,
+          'is-float': 1.35,
+          'is-sci': 1.35e-5,
+          'is-str-null': 'null',
+          'is-str-true': 'true',
+          'is-str-false': 'false',
+          'is-str-int': '1234567',
+          'is-str-float': '1.35',
+          'is-str-sci': '1.35e-5',
+        },
+      ],
+    ])
+  })
+
+  test('bails out on import * as import', () => {
+    expect(
+      findStaticPlugins(js`
+        import * as plugin from './plugin'
+
+        export default {
+          plugins: [plugin]
+        }
+      `),
+    ).toEqual(null)
+  })
+
+  test('bails out on inline plugins', () => {
+    expect(
+      findStaticPlugins(js`
+        import plugin1 from './plugin1'
+
+        export default {
+          plugins: [plugin1, () => {} ]
+        }
+      `),
+    ).toEqual(null)
+
+    expect(
+      findStaticPlugins(js`
+        let plugin1 = () => {}
+
+        export default {
+          plugins: [plugin1]
+        }
+      `),
     ).toEqual(null)
   })
 
   test('bails out on non `require` calls', () => {
     expect(
       findStaticPlugins(js`
-      export default {
-        plugins: [frequire('./plugin1')]
-      }
-    `),
+        export default {
+          plugins: [falserequire('./plugin1')]
```

Comment:
This reads like a typo rather than an intentional non-require call. Maybe like `load` or something?

Reply:
Haha I had it call `frequire` before and I actually went in and fixed it since I thought it's a typo 🤕  agree i will just call it load so there's absolutely no confusion.